mkosi

```diff
@@ -2335,6 +2337,9 @@ def parse_args():
     group.add_argument("--bmap", action='store_true', help='Write block map file (.bmap) for bmaptool usage (only raw_gpt, raw_btrfs)')
     group.add_argument("--password", help='Set the root password')
 
+    group = parser.add_argument_group("Host configuration")
+    group.add_argument("--extra-search-paths", help="List of colon-separated paths to look for programs before looking in PATH")
```

Comment:
hmm, so extra_search_paths is initially a string, right? and then magically becomes a list... I don't like this very much I must say. I think it would be much nicer if we'd maintain it as list all the way.

Also I think we should keep extending the list. i.e. could you please rework this to work like the PackageAction thing, except we split by colon rather than comma for this? It think this would be tons nicer and more systematic.

i.e. it's fine to resolve the env vars in it later, but please let's at least stick to one type for the thing for the whole run

Reply:
Rework to use a list action (added a separate commit as groundwork). See if that is more in tune with what you are thinking.